src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -603,6 +619,35 @@ internal override CSharpAttributeData EarlyDecodeWellKnownAttribute(ref EarlyDec
                 {
                     arguments.GetOrCreateData<ParameterEarlyWellKnownAttributeData>().HasCallerMemberNameAttribute = true;
                 }
+                else if (CSharpAttributeData.IsTargetEarlyAttribute(arguments.AttributeType, arguments.AttributeSyntax, AttributeDescription.CallerArgumentExpressionAttribute))
+                {
+                    // PROTOTYPE(caller-expr): From 333fred review:
+                    // As the language exists today we can do this as an early attribute, but if we do implement the
+                    // nameof proposal (allowing nameof(parameter) in parameters/attributes on methods) I don't believe it
+                    // will end well (this will cause a loop or fail to bind entirely).
+                    // Consider moving binding of the actual attribute data into regular attribute binding.
+                    arguments.GetOrCreateData<ParameterEarlyWellKnownAttributeData>().HasCallerArgumentExpressionAttribute = true;
+
+                    var index = -1;
+                    var attribute = arguments.Binder.GetAttribute(arguments.AttributeSyntax, arguments.AttributeType, out bool generatedDiagnostics);
+                    if (!generatedDiagnostics && !attribute.HasErrors)
```

Comment:
>!generatedDiagnostics [](start = 24, length = 21)

A diagnostics doesn't necessarily mean an error. I guess we could have only resolved warnings there and quite possible for an unresolved diagnostics to be resolved to an warning.

Reply:
@AlekseyTs That sounds reasonable. The tests I have passes regardless this condition, do you have any test case in mind that can get affected so I add it?

Reply:
> That sounds reasonable. The tests I have passes regardless this condition, do you have any test case in mind that can get affected so I add it?

I would remove the ```generatedDiagnostics``` check. We either can get the value from the attribute or we cannot. We should probably also open an issue to look at other usages of this out parameter. However, we should make sure that the DecodeCallerArgumentExpressionAttribute is either unreachable or robust in the face of obvious error scenarios (the value was not supplied, type of the value isn't what we expect, etc). Another option is to adjust GetAttribute to check for presence of resolved errors rather than merely checking for any resolved diagnostics.

---
In reply to: [600291989](https://github.com/dotnet/roslyn/pull/51952#discussion_r600291989) [](ancestors = 600291989)

Reply:
Done.